openedx/xblock-drag-and-drop-v2
# [BB-6301] fix: inconsistent behaviour when showing result

## Description

The PR fixes the following behavior when showing result in assessment mode.

For items with multiple correct target zones, when the `Show Answer` button is pressed, all the items accumulate in one of the zones, even if some of the items were dropped in any other correct zone.

The following logic is implemented with this PR:
1. All items dropped in correct zones by the user, are retained in those zones.
2. An item which is either dropped in an incorrect zone, or is present in the item bar : the item is placed in a zone with no other item or with the least number of items

**Before:**
![Before](https://user-images.githubusercontent.com/36229857/172890145-edd33e20-fc95-4212-b55a-2f81d5edb946.gif)

**After:**
![After](https://user-images.githubusercontent.com/36229857/172890200-87266384-881e-4438-b97e-1b441e6bd7e4.gif)


## Supporting information

OpenCraft internal ticket : [BB-6301](https://tasks.opencraft.com/browse/BB-6301)

## Testing instructions

1. Checkout this branch in <devstack_parent>/src directory in your local setup
2. Login to Studio shell, switch to edxapp virtual env
3. Go to /edx/src/xblock-drag-and-drop-v2 and run pip install -e .
4. Go to Studio UI and create a new course
5. Import this zipped course [drapdrop.tar.gz](https://github.com/openedx/xblock-drag-and-drop-v2/files/8871857/drapdrop.tar.gz)
6. Go to the LMS UI and open the drag and drop unit in the imported course
7. All items with `Red` in name can be dropped in any of the red drop zones and all items `Blue` in name can be dropped in any of te blue drop zones.
8. Place some items in correct zones and other in incorrect zones and submit
9. Press `Show Answer`
10. Verify behaviour as described above.

## Deadline

"None" if there's no rush, or provide a specific date or event (and reason) if there is one.

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.
